test/hotspot/gtest/runtime/test_os.cpp

```diff
@@ -865,3 +866,13 @@ TEST_VM(os, iso8601_time) {
   // Canary should still be intact
   EXPECT_EQ(buffer[os::iso8601_timestamp_size], 'X');
 }
+
+TEST_VM(os, is_first_C_frame) {
+  #ifndef _WIN32
+  frame invalid_frame;
+  EXPECT_TRUE(os::is_first_C_frame(&invalid_frame)); // the frame has zeroes for all values
+
```

Comment:
Please add a test with valid looking but garbage pointers, to test that your safefetch really works. We usually do this by reserving + protecting a stripe of memory and using that one as guaranteed faulting pointer.

Reply:
I thought about, but was unsure how to it properly.

Reply:
No problem, it's probably enough in its current form.

Reply:
Ok, but it would be cool if you could tell me how to do it, because I have the suspicion, that this not the only PR that I will ever write regarding segfaults.

Reply:
But I have to wait till Monday to let someone write an issue :)